rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -267,9 +267,9 @@ protected Void doInBackground(Void... voids) {
             boolean isUserAuthorized = syncUtils.verifyAuthorization();
 
             Activity currentActivity = RDTApplication.getInstance().getCurrentActivity();
-            boolean isFormActivity = currentActivity instanceof RDTJsonFormActivity;
+            boolean isFormSourceActivity = currentActivity instanceof PatientProfileActivity;
 
-            if (!isUserAuthorized && !isFormActivity) {
+            if (!isUserAuthorized && !isFormSourceActivity) {
```

Comment:
And this to `if (!isUserAuthorized && isRegisterActivity)`

Reply:
to only logout when on the register page

Reply:
fixed.